src/qibo/backends/clifford.py

```diff
@@ -671,13 +212,13 @@ def execute_circuit_repeated(self, circuit, initial_state=None, nshots: int = 10
                 Defaults to :math:`1000`.
 
         Returns:
-            (:class:`qibo.quantum_info.clifford.Clifford`): Object giving access to the final results.
+            :class:`qibo.quantum_info.clifford.Clifford`: Object storing to the final results.
         """
         circuit_copy = circuit.copy()
         samples = []
         for _ in range(nshots):
             res = self.execute_circuit(circuit_copy, initial_state, nshots=1)
-            [m.result.reset() for m in circuit_copy.measurements]
+            [measurement.result.reset() for measurement in circuit_copy.measurements]
```

Comment:
Is this line needed? If so, why is it not assigned to anything

Reply:
If I understand correctly, it should be a one-line way to write:
```suggestion
            for measurement in circuit_copy.measurements:
                measurement.result.reset()
```
since `.reset()` has no output, but just modifying the result state.
https://github.com/qiboteam/qibo/blob/428265540de7b5d03de8877eab1f945b9119fe59/src/qibo/measurements.py#L130-L133

However, the two-lines version is clearer, and not much more verbose. I'd suggest that.

Reply:
Btw: in this case you're also creating a list, so allocating memory for it (that will be negligible, and you hope the garbage collector to get rid of it soon). For other cases, consider using generators
```suggestion
            (measurement.result.reset() for measurement in circuit_copy.measurements)
```

But in this case it would not work, because if the generator is not looped, the internal loop does not execute (that's its purpose, you make the loop lazy to avoid memory allocation, required for a loop happening ahead of time).

Reply:
Yes, this resets the measurements for repeated execution. Otherwise, you would generate the samples the first time only and keep using them over and over instead of regenerating them. They are not assigned to anything because there's nothing to be assigned.

Reply:
Then it's the two-line loop that Alessandro suggested is preferred. The suggestion is already committed.